tests/test_modules.hpp

```diff
@@ -0,0 +1,275 @@
+/**
+ * SPDX-FileCopyrightText: Copyright (c) 2021-2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "srf/experimental/modules/segment_modules.hpp"
+#include "srf/segment/builder.hpp"
+
+#include <typeinfo>
+
+namespace srf::modules {
+
+class SimpleModule : public SegmentModule
+{
+  public:
+    SimpleModule(std::string module_name);
+    SimpleModule(std::string module_name, nlohmann::json config);
+
+    void initialize(segment::Builder& builder) override;
+
+    bool m_was_configured{false};
+
+  private:
+    bool m_initialized{false};
+};
+
+SimpleModule::SimpleModule(std::string module_name) : SegmentModule(std::move(module_name)) {}
+
+SimpleModule::SimpleModule(std::string module_name, nlohmann::json config) :
+  SegmentModule(std::move(module_name), std::move(config))
+{}
+
+void SimpleModule::initialize(segment::Builder& builder)
+{
+    VLOG(10) << "MyModule::operator() called for '" << this->name() << "'" << std::endl;
+
+    if (config().contains("simple_key_1"))
+    {
+        m_was_configured = true;
+    }
+
+    /** First linear path **/
+    auto input1 = builder.make_node<bool, unsigned int>(this->get_module_component_name("input1"),
```

Comment:
you are calling this method, because you want the namespaced name of a particular component?

would you rather have the a `ModuleBuilder` that is created from a `Builder + SegmentModule` so that the behavior of `make_node` can be more transparent?

with this you could just have

```
auto input1 = mod_builder.make_node<bool, unsigned int>("input1", ...);
```

Reply:
Calling it like this so we always get a unique name for every local component; but, I like the idea of making this the responsibility of the builder, so this approach will change shortly.